### PR TITLE
KOGITO-177 - Advanced queries for domain and technical objects using …

### PR DESCRIPTION
**Pull request by cristianonicolai**
…GraphQL

### CODE REVIEW

File: kogito-codegen/src/main/java/org/kie/kogito/codegen/process/persistence/proto/ProtoGenerator.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 public interface ProtoGenerator<T> {
     
-    static final String INDEX_COMMENT = "@Field(store = Store.YES, analyze = Analyze.YES)"; 
+    String INDEX_COMMENT = "@Field(store = Store.YES)";
```
[the remark this answers is not in the document]
Comment (cristianonicolai) [reply kogito-codegen/src/main/java/org/kie/kogito/codegen/process/persistence/proto/ProtoGenerator.java]: static final is default anyway for interface

[the remark this answers is not in the document]
Comment (mswiderski) [reply kogito-codegen/src/main/java/org/kie/kogito/codegen/process/persistence/proto/ProtoGenerator.java]: Roger that... as I told you I want to learn something :)

File: data-index/data-index-service/src/main/java/org/kie/kogito/index/graphql/GraphQLScalarTypeProducer.java [hunk 2]
```diff
@@ -33,6 +34,22 @@
 @ApplicationScoped
 public class GraphQLScalarTypeProducer {
 
+    public static ZonedDateTime parseDateTime(String s) {
+        try {
+            return ZonedDateTime.parse(s, DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+        } catch (DateTimeParseException e) {
+            throw new CoercingSerializeException("Invalid RFC3339 value : '" + s + "'. because of : '" + e.getMessage() + "'");
```
[the remark this answers is not in the document]
Comment (cristianonicolai) [reply data-index/data-index-service/src/main/java/org/kie/kogito/index/graphql/GraphQLScalarTypeProducer.java]: done

File: data-index/data-index-storage/data-index-storage-api/src/main/java/org/kie/kogito/index/query/FilterCondition.java [hunk 3]
```diff
@@ -16,17 +16,22 @@
...
-import org.kie.kogito.index.model.UserTaskInstance;
-
-public interface QueryService {
-
-    Collection<ObjectNode> queryDomain(String domain, String query);
-
-    Collection<ProcessInstance> queryProcessInstances(ProcessInstanceFilter filter);
-
-    Collection<UserTaskInstance> queryUserTaskInstances(UserTaskInstanceFilter filter);
+    IN,
```
[the remark this answers is not in the document]
Comment (cristianonicolai) [reply data-index/data-index-storage/data-index-storage-api/src/main/java/org/kie/kogito/index/query/FilterCondition.java]: done

File: data-index/data-index-service/src/main/java/org/kie/kogito/index/graphql/query/GraphQLQueryMapper.java [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+            }
+        }).collect(Collectors.counting()) == type.getFields().size();
+    }
+
+    private Function<Object, Stream<AttributeFilter>> mapRecursiveArgument(String joining, GraphQLQueryParser parser) {
+        return argument -> {
+            Stream<AttributeFilter> stream = ((List) argument).stream().flatMap(args -> parser.apply(args).stream());
+            List<AttributeFilter> filters = stream.collect(toList());
+            switch (joining) {
+                case "and":
```
[the remark this answers is not in the document]
Comment (cristianonicolai) [reply data-index/data-index-service/src/main/java/org/kie/kogito/index/graphql/query/GraphQLQueryMapper.java]: removed these constants

[the remark this answers is not in the document]
Comment (mswiderski) [reply data-index/data-index-service/src/main/java/org/kie/kogito/index/graphql/query/GraphQLQueryMapper.java]: I guess you meant ... "moved to these constants"?

anyway looks great now!

**Comment by mswiderski** [comment]
so from my point of view this is ready to be merged, @nmirasch now final work is on you :)

**Pull request merged**